app/scripts/controllers/detect-tokens.js

```diff
@@ -138,9 +142,13 @@ export default class DetectTokensController {
       });
       this.hiddenTokens = hiddenTokens;
     });
-    preferences.store.subscribe(({ selectedAddress }) => {
-      if (this.selectedAddress !== selectedAddress) {
+    preferences.store.subscribe(({ selectedAddress, useStaticTokenList }) => {
+      if (
```

Comment:
No I think this is right. If the selectedAddress changes the tokens we should detect will change too because you may have balances for different tokens across different addresses

Reply:
@adonesky1 fyi: this originally was `&&`, and was changed in this commit https://github.com/MetaMask/metamask-extension/commit/bbbb2b056e1ea2c5dc12cad83e0551e3ef0cd0ed

My original question was asked when it was `&&`

Reply:
So I agree with your assessment

Reply:
Ah ok! 👍